src/handlers.cpp

```diff
@@ -1893,6 +1933,25 @@ void ImpuRegDataTask::on_sar_response(Diameter::Message& rsp)
   return;
 }
 
+std::string ImpuRegDataTask::search_term()
```

Comment:
I think this should be two functions.
The first one (wildcard_id()?) covers "_hss_wildcard.empty() ? _sprout_wildcard : _hss_wildcard". 
The second (public_id()) then can use this - e.g. std::string wildcard_id = wildcard_id(); return wildcard_id.empty() ? _impu : wildcard_id;"